File: JAVA/Practica8/PROBLEMA_DIAMANTE/Main.java
package JAVA.Practica8.PROBLEMA_DIAMANTE;

import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Valor para x: ");
        int x = scanner.nextInt();
        System.out.print("Valor para y: ");
        int y = scanner.nextInt();
        System.out.print("Valor para z: ");
        int z = scanner.nextInt();
        D objeto = new D(x, y, z);
        System.out.println("--------------------------");
        objeto.x += 1;
        objeto.z += 1;
        System.out.println("IncrementaXZ:");
        System.out.println("x: " + objeto.x + ", y: " + objeto.y + ", z: " + objeto.z);
        System.out.println("--------------------------");
        objeto.y += 1;
        objeto.z += 1;
        System.out.println("\nIncrementaYZ:");
        System.out.println("x: " + objeto.x + ", y: " + objeto.y + ", z: " + objeto.z);
        System.out.println("--------------------------");
        System.out.println("IncrementaXYZ:");
        objeto.incrementaXYZ();
        System.out.println("x: " + objeto.x + ", y: " + objeto.y + ", z: " + objeto.z);
        scanner.close();
    }
}

File: PYTHON/Practica8/PROBLEMA_DIAMANTE/Main.py
from D import D
x = int(input("Valor para x: "))
y = int(input("Valor para y: "))
z = int(input("Valor para z: "))

objeto = D(x, y, z)
print("-----------------------------")
print("IncrementaXZ:")
objeto.incrementaXZ()
print(f"x: {objeto.x}, y: {objeto.y}, z: {objeto.z}")
print("-----------------------------")
print("IncrementaYZ:")
objeto.incrementaYZ()
print(f"x: {objeto.x}, y: {objeto.y}, z: {objeto.z}")
print("-----------------------------")
print("IncrementaXYZ:")
objeto.incrementaXYZ()
print(f"x: {objeto.x}, y: {objeto.y}, z: {objeto.z}")

